src/protocols/http/hpr_http_roaming.erl

```diff
@@ -0,0 +1,549 @@
+%%%-------------------------------------------------------------------
+%%% @author jonathanruttenberg
+%%% @copyright (C) 2022, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 17. Sep 2022 3:40 PM
+%%%-------------------------------------------------------------------
+-module(hpr_http_roaming).
+-author("jonathanruttenberg").
+
+%% Uplinking
+-export([
+    make_uplink_payload/7,
+    select_best/1,
+    gateway_time/1,
+    response_stream/1
+]).
+
+%% Downlinking
+-export([
+    handle_message/1,
+    handle_prstart_ans/1,
+    handle_xmitdata_req/1
+]).
+
+%% Tokens
+-export([
+    make_uplink_token/5,
+    parse_uplink_token/1
+]).
+
+-export([new_packet/3]).
+
+-define(NO_ROAMING_AGREEMENT, <<"NoRoamingAgreement">>).
+
+-define(JOIN1_DELAY, 5_000_000).
+-define(JOIN2_DELAY, 6_000_000).
+-define(RX2_DELAY, 2_000_000).
+-define(RX1_DELAY, 1_000_000).
+
+%% Roaming MessageTypes
+-type prstart_req() :: map().
+-type prstart_ans() :: map().
+-type xmitdata_req() :: map().
+-type xmitdata_ans() :: map().
+
+-type netid_num() :: non_neg_integer().
+-type gateway_time() :: non_neg_integer().
+
+-type downlink() :: {
+    ResponseStream :: hpr_router_stream_manager:gateway_stream(),
+    Response :: any()
+}.
+
+-type transaction_id() :: integer().
+-type region() :: atom().
+-type token() :: binary().
+-type dest_url() :: binary().
+-type flow_type() :: sync | async.
+
+-define(TOKEN_SEP, <<"::">>).
+
+-record(packet, {
+    packet_up :: hpr_packet_up:packet(),
+    gateway_time :: gateway_time(),
+    response_stream :: hpr_router_stream_manager:gateway_stream()
```

Comment:
Maybe we can put this type somewhere else.
The `hpr_router_stream_manager` module serves a fairly specific purpose for the `packet_router` protocol. 
The gateway streams happen to be represented by the same thing right now, but I don't think we should cross mingling the protocol types.

Reply:
...
 Type is now defined in hpr_http_roaming module.